### PR TITLE
Remove bzip2 support #9

### PR DESCRIPTION
**Pull request by kelson42**

### CODE REVIEW

File: src/cluster.cpp [hunk 1]
```diff
@@ -224,20 +219,6 @@ namespace zim
...
-      case zimcompBzip2:
-        {
-#if defined(ENABLE_BZIP2)
-          log_debug("uncompress data (bzip2)");
-          zim::Bunzip2Stream is(in);
-          is.exceptions(std::ios::failbit | std::ios::badbit);
-          read_header(is);
-          read_content(is);
-#else
-          throw std::runtime_error("bzip2 not enabled in this library");
```
Comment (mgautierfr) [inline src/cluster.cpp]: I would have keep this to properly handle a potential zim file with bzip2 compression.
Maybe changing a bit the error message and moving the `case` at the end of the `switch` but while keeping the same behavior (throw a std::runtime_error)

File: src/cluster.cpp [hunk 2]
```diff
@@ -286,20 +267,6 @@ namespace zim
...
-      case zimcompBzip2:
-        {
-#if defined(ENABLE_BZIP2)
-          log_debug("compress data (bzip2)");
-          zim::Bzip2Stream os(out);
-          os.exceptions(std::ios::failbit | std::ios::badbit);
-          clusterImpl.write(os);
-          os.end();
-#else
-          throw std::runtime_error("bzip2 not enabled in this library");
```
Comment (mgautierfr) [inline src/cluster.cpp]: Same here.

File: src/tools/zimDump.cpp [hunk 3]
```diff
@@ -289,7 +289,6 @@ void ZimDumper::listArticle(const zim::Article& article, bool extra)
         case zim::zimcompDefault: std::cout << "default"; break;
         case zim::zimcompNone:    std::cout << "none"; break;
         case zim::zimcompZip:     std::cout << "zip"; break;
-        case zim::zimcompBzip2:   std::cout << "bzip2"; break;
```
Comment (mgautierfr) [inline src/tools/zimDump.cpp]: Same here. Even if we totally drop the bzip2 support, if we found a zim with bzip2 compression, we know this is a bzip2 compression. It is better than `unknown`.

We may change the displayed text to `"bzip2 (not supported)"`

**Comment by mgautierfr** [comment]
Why have you closed this PR ?
There are changes to make before merging it but I can keep it open.

File: src/cluster.cpp (hunk 1 above)
Comment (kelson42) [inline src/cluster.cpp]: yes

File: src/cluster.cpp (hunk 2 above)
Comment (kelson42) [inline src/cluster.cpp]: yes

File: src/tools/zimDump.cpp (hunk 3 above)
Comment (kelson42) [inline src/tools/zimDump.cpp]: yes

**Comment by kelson42** [comment]
@mgautierfr Ready for second round review

**Comment by mgautierfr** [comment]
Can you rebase your PR on master ?

Your PR is still based on a old master who try to get dependencies with ssh.
Now that I've removed ssh key access in travis, the travis build system fails.

**Comment by mgautierfr** [comment]
This has been correctly merged in #22.

**Pull request merged**